src/profiler.h

```diff
@@ -234,6 +224,16 @@ class Profiler {
         _instance.addRuntimeStub(address, length, name);
     }
 
+    static void JNICALL ThreadStart(jvmtiEnv* jvmti, JNIEnv* jni, jthread thread) {
+        _instance._thread_names.update(thread);
+        _instance._engine->onThreadStart();
```

Comment:
`_engine` can be concurrently deleted, right?

Reply:
Hm, yes. Is it ok to add yet another mutex to give thread safe access to `_engine`? Or can we just ignore possible data race and check that `_engine != NULL` ?

Reply:
Data race is not a problem. But unfortunately `_engine != NULL` does not help, since `_engine` can be deleted by another thread right after this check,